contracts/StrategyCvxCrvHelper.sol

```diff
@@ -263,14 +280,20 @@ contract StrategyCvxCrvHelper is
             }
         }
 
-        // 3. Sell CVX -> CRV
+        // 3. Lock CVX for CRV
         uint256 cvxTokenBalance = cvxToken.balanceOf(address(this));
+
         if (cvxTokenBalance > 0) {
-            _swapExactTokensForTokens(
-                sushiswap,
-                cvx,
-                cvxTokenBalance,
-                getTokenSwapPath(cvx, crv)
+            uint256 prevTreeBalance = CVX_VAULT.balanceOf(BADGER_TREE);
+            cvxToken.safeApprove(address(CVX_VAULT), 0); //Reset to avoid dust
+            cvxToken.safeApprove(address(CVX_VAULT), cvxTokenBalance); // Approve exact amount
```

Comment:
Can probably be moved to a separate function instead of being called on each harvest

Reply:
I'd hate to do this and then have the coordination circus to patch before harvest

Reply:
Can we vote it?

Reply:
@shuklaayush, if we are approving the specific harvested amounts why would you want to move it to a separate function? Or do you mean having a function to approve to MAX?

Reply:
Yep, I meant having a separate function where we just approve the max once. `bveCvx` is controlled by us so there's no real risk imo

Reply:
I that case I think it makes sense @GalloDaSballo, we will have to call it once as we have done before with patchPaths, etc. Any gas we can save on harvest is worth it.

Reply:
After experimenting, there seems to be no meaningful difference.

Before eip 1559 my approach would save gas as it triggered over 30k in gas refunds, as of now the refunds should be around 6k gas